src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsImpl.java

```diff
@@ -102,7 +102,7 @@ public String getUsername() {
 
     @Override
     public boolean isUsernameSecret() {
-        return usernameSecret;
+        return Boolean.TRUE.equals(usernameSecret);
```

Comment:
💡 FYI `Unboxing of 'usernameSecret' may produce 'NullPointerException'` is a false positive here as it's not possible to have that value being null, due to the `readResolve` method.

But still OK to have this change.

Reply:
I see. However, the field is annotated as nullable and there's setter accepting `null`, so in theory a NPE is possible.

Reply:
If you mean: https://github.com/jenkinsci/credentials-plugin/blob/a2b5c5fba5b7fe77854efeab539b5e4843d52719/src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsImpl.java#L108-L111

It does not accept `null` as it's the boolean primitive.

Reply:
:man_facepalming: …

Reply:
Revert this misleading change and use `@SuppressFBWarnings` if necessary, though IIUC it should _not_ be necessary since I marked the field `@Nullable` not `@CheckForNull` to begin with.